filterByTag.js

```diff
@@ -0,0 +1,82 @@
+class Filter {
+  constructor(tagSelector) {
+    this.tagSelector = document.querySelectorAll(tagSelector); // What tag to filter
```

Comment:
When this class is used on line 62, no value is passed for the `tagSelector`, so it will be `undefined` here and `querySelector()` will return `null`.

That's not a problem though, because nothing happens to `tagSelector` after this point. It's basically dead code, and I think the entire constructor could have been removed.